contracts/V1/DataTypes.sol

```diff
@@ -3,22 +3,21 @@ pragma solidity ^0.8.8;
 //Just so typechain generates types for the structs bellow
 contract DataTypes {
     constructor(
-        Transaction memory a,
-        Block memory b,
-        SignedBlock memory c,
-        JoinChannel memory d,
-        SignedJoinChannel memory e,
-        JoinChannelAgreement memory f,
-        ConfirmedJoinChannelAgreement memory g,
-        LeaveChannel memory h,
-        LeaveChannelAgreement memory i
+        Block memory a,
+        SignedBlock memory b,
+        BlockConfirmation memory c,
+        Transaction memory d,
+        TransactionHeader memory e,
```

Comment:
When a type is contained within a type (TxHeader and TxBody are within Tx) it's enough to only list the top level type (Tx), but this is not important you can list all - this is just so typechain generates the types and they're not skipped by the compiler